### ISSUE TITLE
Error updateing offer - missing parameter

### ISSUE THREAD

**Issue by ZajdaBear** [allegro/allegro-api#10435]
Hi, we have sent a request to PATCH offer 16154692985, and we get an error
````
{
    "errors": [
        {
            "code": "ConstraintViolationException.MissingRequiredParameters",
            "message": "Missing required parameters: 129629",
            "details": "ConstraintViolationException.MissingRequiredParameters",
            "path": "parameters",
            "userMessage": "Missing mandatory parameters: Handle type.",
            "metadata": {}
        }
    ]
}
````
however when I check the offer using GET request, I can see mentioned parameter present 

````
{
                        "id": "129629",
                        "name": "Typ rukojeti",
                        "values": [
                            "ostatní"
                        ],
                        "valuesIds": [
                            "129629_3"
                        ],
                        "rangeValue": null
                    }
````

Can you please advice?

PATCH < trace-id: f99df96050539c35

**Comment by MaciejFrackowiak** [comment]
To your offer 16154692985 there is assigned a product with id=42a7d56a-b0f0-4b11-8676-0f2473fdb927. If you retrieve its details (GET /sale/products/{productId}, you will find that the product doesn't have the required parameter "129629" (Typ rukojeti). Moreover, parameter "129629" is a product parameter, so it means that describes a product. Whenever you edit an offer with PATCH /sale/product-offers/{offerId}, you should remember to add a value of a parameter "129629" to your request - to the parameter section under productSet.

**Comment by ZajdaBear** [comment]
Hi, thanks for the response. I do not fully understand, why I need to add the product parameter to the PATCH request, when it is already present on the offer. I am not changing product parameters on the offer. My request looks like this
````
{
    "productSet": [
        {
            "product": {
                "id": "4015671554352",
                "idType": "GTIN"
            },
            "quantity": {
                "value": 1
            }
        }
    ],
    "delivery": {
        "handlingTime": "P2D",
        "shippingRates": {
            "id": "c680eb30-242a-40a4-855d-d8fc82300ad1"
        }
    },
    "publication": {
        "status": "ACTIVE"
    },
    "language": "cs-CZ",
    "afterSalesServices": {
        "impliedWarranty": {
            "id": "4d060e7b-19c1-4173-9e62-607ddb77bec9"
        },
        "returnPolicy": {
            "id": "a4693c8c-ade1-4969-bdce-3e39169f252c"
        }
    },
    "name": "SLOUPOVÁ VRTAČKA GSB 32",
    "sellingMode": {
        "format": "BUY_NOW",
        "price": {
            "amount": "16671.00",
            "currency": "CZK"
        }
    },
    "stock": {
        "available": 2,
        "unit": "UNIT"
    },
    "location": {
        "city": "Šestajovice",
        "countryCode": "CZ",
        "postCode": "25092"
    },
    "external": {
        "id": "55435"
    },
    "messageToSellerSettings": {
        "mode": "HIDDEN"
    },
    "taxSettings": {
        "rates": [
            {
                "rate": "21.00",
                "countryCode": "CZ"
            }
        ]
    }
}

**Comment by MaciejFrackowiak** [comment]
You have to add it because the assigned product doesn't have it in its structure.

**Comment by ZajdaBear** [comment]
Hi, can you double check this is a correct behavior? I just tested on the sandbox, and I was able to patch the offer, that was missing parameters (eg. 215930) If I look on the attached product (aa3eaa61-d283-4777-a332-5b02c0d53cdc), it still misses the parameter 215930, but the PATCH request works fine, as I would expect it to, but it does not align with what you previously claimed. offer id on sandbox:  7775343841

**Comment by MaciejFrackowiak** [comment]
Yes, I have checked your requests. When you edited an offer on production you provided the id of the product (as a GTIN code). On Sandbox you made several requests - every single one which referred to the product finished with the error and information about missing parameters. Only requests which had product safety information finished without errors (they didn't have a product.id in its structure).

**Comment by ZajdaBear** [comment]
Thanks, can you please provide me with clear rules, on when I need to provide required parameters, on offer PATCH request (when assigned product is missing those)? I still believe, this behavior is not correct, as offer has all requested parameters, I ma not updating parameters, yet still I receive missing parameters error.

Cheers!

Jirka Z

**Comment by MaciejFrackowiak** [comment]
You receive errors because, during the update, we check if the product has all of its parameters. As I mentioned before, not all products from our product catalog have all the required parameters. Always pay attention to whether product parameters are set - if yes, you don't have to provide them.
